src/accelerate/utils/launch.py

```diff
@@ -537,7 +537,7 @@ def __call__(self, index, *args):
         ):
             # Prepare the environment for torch.distributed
             os.environ["LOCAL_RANK"] = str(index)
-            os.environ["RANK"] = str(index)
+            os.environ["RANK"] = str(int(os.environ["NPROC"]) * int(os.environ["NODE_RANK"]) + index)
```

Comment:
Can we assume that `NPROC` and `NODE_RANK` will always be set? When using `notebook_launcher`, that should be true, but AFAICT, `PrepareForLaunch` can also be used in other contexts, e.g. in `debug_launcher` and maybe in some custom user code. So would it be better not to assume the presence of these env vars and have defaults for them if they don't exist? For example `nproc = int(os.environ.get("NPROC", 1))`.